### DISCUSSION TITLE
Is error.tsx's reset function pointless if it doesn't retry fetch requests?

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#49935] [App Router]
Error pages can receive a `reset` callback to try to rerender the erroring page:
https://nextjs.org/docs/app/building-your-application/routing/error-handling#recovering-from-errors

But this **doesn't seem to execute any server-side requests again**. Doesn't this make this whole functionality pretty pointless, considering that most errors are probably caused by failed network requests?

**Comment by SanderCokart** [comment]
I am just thinking the same thing...

**Comment by gerkim62** [comment]
## Refreshing the Route Using Next.js Router Hook

Instead of using `reset()`, you can simply use the router hook to refresh the route in Next.js. Here's how you can do it:

```jsx
// Import necessary dependencies
'use client'
import { useRouter } from 'next/navigation'

// Define the Error component
export default function Error({
  error,
  reset,
}: {
  error: Error & { digest?: string }
  reset: () => void
}) {
  // Access the router hook
  const router = useRouter()

  // Render the component
  return (
    <div>
      <h2>Something went wrong!</h2>
      
      {/* use refresh instead of reset here */}
      <button onClick={() => router.refresh()}>Try again</button>
    </div>
  )
}

**Comment by SanderCokart** [comment]
Hehe funny thing actually but that doesn't refetch in my testing. On an
error page only a window.location.reload works.

**Comment by LutherTS** [comment]
Refresh didn't work for me either, so I'm just using  const router = useRouter() router.back() instead to see the user back to the previous page.

**Comment by Atliac** [comment]
`startTransition()` and `router.refresh()` come to the rescue.

https://github.com/vercel/next.js/issues/63369#issuecomment-2211698627

**Reply by musjj** [reply]
My setup looks like this:

```tsx
// Server component
export default Page() {
  return (
    <Suspense>
      <DataFetchingComponent /> {/* Client component */}
    </Suspense>
  );
}
```

When `DataFetchingComponent` errors, even calling `router.refresh()` in `error.tsx` does not seem to cause that component to refetch. Is there a way to deal with this?

**Reply by gerkim62** [reply]
i would suggest doing a `window.location.reload()` if the application is not too heavy.
